app/views/helpers/Title.scala

```diff
@@ -18,14 +18,28 @@ package views.helpers
 
 import play.api.i18n.Messages
 
-case class Title(headingKey: String, sectionKey: String = "", headingArg: String = "", headingArgs: Option[Seq[String]] = None) {
+case class Title(
+  headingKey: String,
+  sectionKey: String = "",
+  headingArg: String = "",
+  headingArgs: Option[Seq[String]] = None,
+  hasErrors: Boolean = false
+) {
 
   def toString(implicit messages: Messages): String = {
     def args = headingArgs.getOrElse(Seq(headingArg))
     if (sectionKey.isEmpty) {
-      messages("title.format", messages(headingKey, args: _*), messages("service.name"))
+      if (hasErrors) {
+        messages("title.hasErrors.format", messages(headingKey, args: _*), messages("service.name"))
+      } else {
+        messages("title.format", messages(headingKey, args: _*), messages("service.name"))
+      }
     } else {
-      messages("title.withSection.format", messages(headingKey, args: _*), messages(sectionKey), messages("service.name"))
+      if (hasErrors) {
```

Comment:
Even better...
```
val key = if (hasErrors) ".hasErrors" else "" // This could be used for both `if` test
messages(s"title$key.withSection.format", messages(headingKey, args: _*), messages("service.name"))
```